torch/_dynamo/variables/builder.py

```diff
@@ -2804,6 +2809,8 @@ def update_dim2constraint(dim, constraint_range, name):
 
         if marked_unbacked:
             dynamic_size = DimDynamic.SIZE_LIKE_UNBACKED
```

Comment:
why did you need to introduce STRICT_SIZE_LIKE_UNBACKED?

Reply:
To preserve backwards compatibility when non strict.

Reply:
Actually on mobile I didn't see your comment. So marked_strict_unbacked is actually derived from STRICT_SIZE_LIKE_UNBACKED

Reply:
mm I am not sure I understand
```
 marked_strict_unbacked = i in getattr(
            e, "_dynamo_strict_unbacked_indices", set()
        )
 ```
 ...
 ```
 elif marked_strict_unbacked:
                constraint_size = RelaxedUnspecConstraint(warn_only=False)
 ```

Reply:
we never read             DimDynamic.STRICT_SIZE_LIKE_UNBACKED, except at line 4421 ored with SIZE_LIKE_UNBACKED sounds not needed.

also we do not have DYNAMIC and STRICT_DYNAMIC, so lets be consistent with both?

Reply:
Sorry I misunderstood, I'll remove it now